src/main/scala/org/broadinstitute/hail/expr/Type.scala

```diff
@@ -97,6 +108,10 @@ abstract class Type extends BaseType {
     sb.append(toString)
   }
 
+  def compact(sb: StringBuilder, printAttrs: Boolean = false) {
```

Comment:
I would have made `compact: Boolean` an flag to `pretty`.  This way is a bit redundant.